op-bindings/bindgen/fixtures_test.go

```diff
@@ -33,15 +146,267 @@ var removeDeploymentSaltTestsFailures = []struct {
 	expectedError  string
 }{
 	{
-		"TestRemoveDeploymentSalt Failure Case #1 Invalid Regex",
+		"Failure Case #1 Invalid Regex",
 		"0x1234abc",
 		"[invalid-regex",
 		"failed to compile regular expression",
 	},
 	{
-		"TestRemoveDeploymentSalt Failure Case #2 Salt Not Found",
+		"Failure Case #2 Salt Not Found",
 		"0x1234abc",
 		"4567",
 		"expected salt: 4567 to be at the beginning of the contract initialization code: 0x1234abc, but it wasn't",
 	},
 }
+
+var compareBytecodeWithOpTests = []struct {
+	name                  string
+	contractMetadataEth   remoteContractMetadata
+	compareInitialization bool
+	compareDeployment     bool
+}{
+	{
+		name: "Safe_v130 Compare Init and Deployed Bytecode",
+		contractMetadataEth: remoteContractMetadata{
+			remoteContract: remoteContract{
+				Name:     "Safe_v130",
+				Verified: true,
+				Deployments: deployments{
+					Op:  common.HexToAddress("0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552"),
+					Eth: common.HexToAddress("0x69f4D1788e39c87893C980c06EdF4b7f686e2938"),
```

Comment:
Looks like all tests here have deploys for both OP and ETH—do we have any tests where the code is only on OP or only on ETH?

Reply:
No, I can add those, thank you!

Reply:
Added new error if there's no deployment on OP

```go
var zeroAddress common.Address
	if contractMetadataEth.Deployments.Op == zeroAddress {
		return fmt.Errorf("no deployment address on Optimism provided for %s", contractMetadataEth.Name)
	}
```

This method assumes you already have the bytecode for the ETH deployment and you're comparing it against OP, so I've added errors if either is not provided

```go
if contractMetadataEth.InitBin == "" {
		return fmt.Errorf("no initialization bytecode provided for ETH deployment for comparison")
	}
	if contractMetadataEth.DeployedBin == "" {
		return fmt.Errorf("no deployed bytecode provided for ETH deployment for comparison")
	}
```

Reply:
I split `compareBytecodeWithOp` into `compareInitBytecodeWithOp` and `compareDeployedBytecodeWithOp` and added checks that the bytecode for ETH was provided, and that a deployment address for OP was provided